### PR TITLE
Make Remarks required for CAU Unit Note type.

### PR DESCRIPTION
**Pull request by dimak1**
*Description of changes:*
- As per submission rules, Remarks are required for a Notice of Caution.
- Update unit tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by dimak1** [comment]
@doug-lovett I went ahead with API spec and made Remarks required for CAU. I asked Liz to update the requirements spreadsheet, as there it shows that Remarks are optional.

File: ppr-ui/src/components/common/Remarks.vue [hunk 1]
```diff
@@ -22,12 +22,12 @@
...
-              :rules="maxLength(420)"
-              name="name"
-              counter="420"
-              label="Remarks (Optional)"
+              :counter="remarksMaxLength"
+              :rules="maxLength(remarksMaxLength)"
+              :label="label"
               class="pl-1"
               data-test-id="remarks-textarea"
+              :error-messages="showRemarksRequiredMsg ? 'Enter remarks' : ''"
```
Comment (dimak1) [inline ppr-ui/src/components/common/Remarks.vue:30]: For some reason, I couldn't get required error to trigger via `:rules` prop, so had to make this work-around.

File: ppr-ui/src/components/common/Remarks.vue [hunk 2]
```diff
@@ -54,15 +55,24 @@ export default defineComponent({
...
+      default: false
     }
   },
   setup (props, { emit }) {
     const { maxLength } = useInputRules()
 
+    const remarksForm = ref(null) as FormIF
+
     const localState = reactive({
+      remarksMaxLength: 420,
```
Comment (dimak1) [inline ppr-ui/src/components/common/Remarks.vue:70]: Made this into a variable as it will be dynamic in upcoming tickets.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1447-lsipnjie.web.app

**Review by cameron-eyds — APPROVED** [review]

**Comment by cameron-eyds** [comment]
LGTM!

**Review by doug-lovett — APPROVED** [review]
Looks good.

File: ppr-ui/src/components/common/Remarks.vue [hunk 3]
```diff
@@ -54,15 +55,24 @@ export default defineComponent({
...
+    },
+    isRequired: {
+      type: Boolean,
+      default: false
     }
   },
   setup (props, { emit }) {
     const { maxLength } = useInputRules()
 
+    const remarksForm = ref(null) as FormIF
```
Comment (orelbn) [inline ppr-ui/src/components/common/Remarks.vue:67]: I think you need to return remarks form.

File: ppr-ui/src/components/common/Remarks.vue (hunk 1 above)
Comment (orelbn) [reply ppr-ui/src/components/common/Remarks.vue:30]: I think you need to return remarks form

**Comment by dimak1** [comment]
Closing this as the Remarks are optional now in API as well as UI.

File: ppr-ui/src/components/common/Remarks.vue (hunk 3 above)
Comment (dimak1) [reply ppr-ui/src/components/common/Remarks.vue:67]: We probably do not need the ref at all.

**Pull request closed without merging**